src/DaemonExec.ts

```diff
@@ -0,0 +1,190 @@
+import * as Docker from 'dockerode'
+
+interface ContainerModem {
+	demuxStream(
+		stream: NodeJS.ReadWriteStream,
+		stdoutStream: NodeJS.WritableStream,
+		stderrStream: NodeJS.WritableStream,
+	): void
+}
+
+interface DockerModem {
+	followProgress(
+		stream: NodeJS.ReadableStream,
+		onFinished: (err?: Error) => Promise<void>,
+		onProgress: (event: {
+			status: string
+			progressDetail: { current: number; total: number }
+			progress: string
+			id: string
+		}) => void,
+	): void
+}
+
+export interface VolumeDefinition {
+	fromPath: string
+	toPath: string
+}
+
+/**
+ * Returns information about the images
+ * @param docker - The docker daemon to pull information from
+ * @returns an array of ImageInfo
+ */
+export async function listImages(docker: Docker): Promise<Docker.ImageInfo[]> {
+	const images = await docker.listImages({})
+	return images
```

Comment:
why not just `return docker.listImages({})`?

Reply:
It was a legacy from debugging while making the code.